crengine/src/private/lvfreetypeface.cpp

```diff
@@ -2194,34 +2231,33 @@ int LVFreeTypeFace::DrawTextString(LVDrawBuf *buf, int x, int y, const lChar32 *
                     printf("regular g%d>%d: ", hg, hg2);
                 #endif
                 // Draw glyphs of this same cluster
-                int prev_x = x;
+                int prev_x_26_6 = x_26_6;
                 for (i = hg; i < hg2; i++) {
                     LVFontGlyphCacheItem *item = getGlyphByIndex(glyph_info[i].codepoint);
                     if (item) {
-                        int w = FONT_METRIC_TO_PX(glyph_pos[i].x_advance);
                         #ifdef DEBUG_DRAW_TEXT
                             printf("%x(x=%d+%d,w=%d) ", glyph_info[i].codepoint, x,
                                     item->origin_x + FONT_METRIC_TO_PX(glyph_pos[i].x_offset), w);
                         #endif
-                        buf->Draw(x + item->origin_x + FONT_METRIC_TO_PX(glyph_pos[i].x_offset),
+                        buf->Draw(FONT_METRIC_TO_PX(x_26_6 + glyph_pos[i].x_offset) + item->origin_x,
```

Comment:
Just asking: at which moment did it feel necessary for you to switch to _26_6 units ? Would things have been bad if you had kept using/rounding to pixels ?

(Not that I would like this and making things even more complicated :), but:) then could item->origin_x also better be stored as 26_6 units ? We use `item->origin_x = (lInt16) slot->bitmap_left;`, which is not in 26_6 units, but I think bitmap_left is just a shortcut value, and the 26_6 unit is probably available in some other slot metric field)

Reply:
> Just asking: at which moment did it feel necessary for you to switch to _26_6 units ? Would things have been bad if you had kept using/rounding to pixels ?

When I decided to add intercharacter spacing for synthetic font weights to avoid rounding errors.

> (Not that I would like this and making things even more complicated :), but:) then could item->origin_x also better be stored as 26_6 units ? We use `item->origin_x = (lInt16) slot->bitmap_left;`, which is not in 26_6 units, but I think bitmap_left is just a shortcut value, and the 26_6 unit is probably available in some other slot metric field)

May be, I have not studied this question yet, so much I had to learn this week ...

Reply:
> When I decided to add intercharacter spacing for synthetic font weights to avoid rounding errors.

And these rounding errors where really visible/noticeable ? Or it was just intellectually to do the right thing ?

(The other question was also purely intellectual - no need to bother if there's no visible glitch that would make it feel it's needed.)

Reply:
> And these rounding errors where really visible/noticeable ? Or it was just intellectually to do the right thing ?

Of course I compared, the difference is obvious. You can compare too.